### PR TITLE
[addons] add addon network functions to new way

### PR DESCRIPTION
**Pull request by AlwinEsch**
This change network related addon to kodi callback functions to new
C++ style.

Related doxygen documentation is already added there.

## Types of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] My code follows the [Code guidelines](https://codedocs.xyz/xbmc/xbmc/code_guidelines.html) of this project 
- [ ] My change requires a change to the documentation, either Doxygen or wiki
- [ ] I have updated the documentation accordingly
- [ ] I have read the [CONTRIBUTING](https://github.com/xbmc/xbmc/blob/master/CONTRIBUTING.md) document
- [ ] I have added tests to cover my change
- [ ] All new and existing tests passed

### CODE REVIEW

**Comment by AlwinEsch** [comment]
Note: After https://github.com/xbmc/xbmc/pull/12187 is in have this conflicts (need to do a update then)

File: xbmc/addons/kodi-addon-dev-kit/include/kodi/Network.h [hunk 1]
```diff
@@ -0,0 +1,189 @@
...
+
+//==============================================================================
+///
+/// \defgroup cpp_kodi_network  Interface - kodi::network
+/// \ingroup cpp
+/// @brief **Network functions**
+///
+/// The network module offers functions that allow you to control it.
+///
+/// These are pure static functions them no other initialization need.
```
Comment (MartijnKaijser) [inline xbmc/addons/kodi-addon-dev-kit/include/kodi/Network.h]: them?

**Review by ksooo — APPROVED** [review]
Code is looking good.

File: xbmc/addons/kodi-addon-dev-kit/include/kodi/Network.h (hunk 1 above)
Comment (AlwinEsch) [inline xbmc/addons/kodi-addon-dev-kit/include/kodi/Network.h]: Have removed the string complete, this description here becomes improved during completion C++ addon documentation.

**Comment by AlwinEsch** [comment]
jenkins build this please

**Comment by AlwinEsch** [comment]
jenkins build this please

**Pull request merged**